.github/workflows/rustc-pull.yml

```diff
@@ -0,0 +1,55 @@
+name: rustc-pull
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run at 04:00 UTC every Monday
+    - cron: '0 4 * * 1'
+
+jobs:
+  pull:
+    if: github.repository == 'rust-lang/rustc-dev-guide'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # We need the full history for josh to work
+          fetch-depth: '0'
+      - name: Install stable Rust toolchain
+        run: rustup update stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: "josh-sync"
+          # Cache the josh directory with checked out rustc
+          cache-directories: "/home/runner/.cache/rustc-dev-guide-josh"
+      - name: Install josh
+        run: RUSTFLAGS="--cap-lints warn" cargo +stable install josh-proxy --git https://github.com/josh-project/josh --tag r24.10.04
+      - name: Setup bot git name and email
+        run: |
+          git config --global user.name 'The rustc-dev-guide Cronjob Bot'
+          git config --global user.email 'rustc-dev-guide@cron.bot'
```

Comment:
Maybe best not to use an email that we don't control.

Reply:
I copied this from [miri](https://github.com/rust-lang/miri/blob/master/.github/workflows/ci.yml#L133), it has been using a similar e-mail address for a similar workflow for a long time. Do you see any attack vector? It's just a string that gets appended to the commit, it doesn't really mean anything.

Reply:
I don't see an obvious one, but to me it seems like good practice to avoid using addresses from unrelated domains. Perhaps you could use `@example.com` instead? At least then it's a known fake domain.

Reply:
Or we could use `github-actions@github.com` which is widely used in GHA automation (like [this](https://github.com/rust-lang/rust/blob/7648efaf1a26aaec38d3babfdd26df1bc8648222/.github/workflows/dependencies.yml#L126-L127)).

Reply:
Good point, changed it to the `github.com` address.